### PR TITLE
Rational transient clean

### PR DESCRIPTION
**Pull request by mloubout**
This branch implement rational transient functions with pre-factorization of this functions to reduce the computational cost.
Contains high-order FD scheme for cross-derivaives in tti_operators.py

### CODE REVIEW

**Comment by mloubout** [comment]
Next step is to remove all this definitions of damping etc from the setup scrip and setup the tti_operator the way the acoustic operator was done to avoid having huge example script with thing people shouldn't have to worry about.
We also need the equivalent of test_adjointA.py for TTI.

**Comment by mloubout** [comment]
@mlange05 @navjotk 
Please review and merge

**Comment by vincepandolfo** [comment]
A lot of commits in this PR are already in master. It'd be better to avoid duplication in git history and removing them

**Comment by mloubout** [comment]
How do I do that?

**Comment by vincepandolfo** [comment]
`git rebase -i master` should be it

File: devito/operators.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from jit_manager import JitManager
+from propagator import Propagator
+import os
+
+
+class Operator(object):
```
Comment (mlange05) [inline devito/operators.py]: This file has been renamed in master to `operator.py`. I don't think this re-introduction is intended.

Comment (mloubout) [inline devito/operators.py]: @pvelesko Any insight on this?

Comment (mloubout) [inline devito/operators.py]: It seems like it it not used anywhere, probably a reminiscence of the old one we din't remove. I'll remove it.

Comment (pvelesko) [inline devito/operators.py]: yeah just trash.

**Comment by mloubout** [comment]
@pvelesko 
Please fix flake8 and typos in your commit and we should be good to go (for review) for this one.
@mlange05 Should we make a new branch to combine your symbolic things with TTI and rewrite the operators and examples accordingly. There is currently too much difference with Acoustic (a lot of things are declared in the example like damp and pad to make it easy while it should be in the operator) and all parameter (epsilon,delta,theta,phi,m) should be part of "model".

**Comment by mlange05** [comment]
Yes, I think switching TTI to the new symbolics should be a separate branch PR.

**Comment by mloubout** [comment]
OK, I need this one to be merged first and I'll get on it.

**Comment by mloubout** [comment]
@pvelesko 
What did you do to the PR? We are back at having conflicts.

**Comment by pvelesko** [comment]
I tried to make a small change in the compiler file but something weird happened because I'm not even seeing that commit anymore.

**Comment by mloubout** [comment]
Ok, don't touch this branch anymore, do your change on your new one and make a different PR once it's done

**Comment by vincepandolfo** [comment]
I guess the conflicts are caused by PR 42 just being merged into master

**Comment by mloubout** [comment]
@mlange05 @navjotk 
This one seems good to go when you're ok with it.

**Comment by mloubout** [comment]
Most of the conflict come from operator and propagator that have complete different argument list and __init__ , this one is gonna be a pain to merge so either by hand or wait until my new branch (tti_clean) is done which will have both this one and current master by hand merge

**Pull request closed without merging**